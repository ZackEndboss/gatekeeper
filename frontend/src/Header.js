import React from 'react';
import PropTypes from 'prop-types';
import R from "ramda";
import {Nav, Navbar, NavItem} from "react-bootstrap";

export function HeaderLinks({links}) {
    return (
        <Nav>
            {R.addIndex(R.map)((link, index) => <NavItem key={index} href="#">{link}</NavItem>)(links)}
        </Nav>
    )
}

export class Header extends React.Component {
    render() {
        const {links} = this.props;
        return (
            <Navbar fluid>
                <Navbar.Header>
                    <Navbar.Brand>
                        Gatekeeper
                    </Navbar.Brand>
                    <HeaderLinks links={links}/>
                </Navbar.Header>
            </Navbar>
        );
    }
}

Header.propTypes = {
    links: PropTypes.array.isRequired
};

export default (Header);
