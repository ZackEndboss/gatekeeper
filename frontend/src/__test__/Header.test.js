import React from 'react';
import renderer from 'react-test-renderer';
import {Header} from '../Header';

jest.mock('react-router-dom/Link', () => 'Link');
jest.mock('react-router-dom/NavLink', () => 'NavLink');

describe('Header component', () => {
    it('render', () => {
        const component = renderer.create(<Header location="group"/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
