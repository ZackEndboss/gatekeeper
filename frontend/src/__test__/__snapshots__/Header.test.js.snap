// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render 1`] = `
<nav
  className="navbar navbar-default"
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <span
        className="navbar-brand"
      >
        Gatekeeper
      </span>
      <ul
        className="nav navbar-nav"
        role={null}
      >
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
          >
            link1
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
          >
            link2
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
          >
            link3
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
